src/components/structures/GroupView.js

```diff
@@ -0,0 +1,134 @@
+                phase: "GroupView.DISPLAY",
+                summary: res,
+            });
+        }, (err) => {
+            this.setState({
+                phase: err.httpStatus == 404 ? "GroupView.NOT_FOUND" :"GroupView.ERROR",
+                summary: null,
+                error: err,
+            });
+        });
+    },
+
+    render: function() {
+        const BaseAvatar = sdk.getComponent("avatars.BaseAvatar");
+        const Loader = sdk.getComponent("elements.Spinner");
+
+        if (this.state.phase == "GroupView.LOADING") {
+            return <Loader />;
+        } else if (this.state.phase == "GroupView.DISPLAY") {
+            const summary = this.state.summary;
+            let avatar_url = null;
+            if (summary.profile.avatar_url) {
+                avatar_url = MatrixClientPeg.get().mxcUrlToHttp(summary.profile.avatar_url);
+            }
+            let description = null;
+            if (summary.profile.long_description) {
+                description = sanitizedHtmlNode(summary.profile.long_description);
+            }
+            return (
+                <div style={{maxWidth: "960px", width: "100%", "margin": "20px auto"}}>
+                    <div className="mx_RoomHeader">
+                        <div className="mx_RoomHeader_wrapper">
+                            <div className="mx_RoomHeader_avatar">
+                                <BaseAvatar url={avatar_url} name={summary.profile.name}
+                                    width={48} height={48} />
+                            </div>
+                            <div className="mx_RoomHeader_info">
+                                <div className="mx_RoomHeader_name">
+                                    <span>{summary.profile.name}</span>
+                                    <span style={{
+                                        fontWeight: "normal",
+                                        fontSize: "initial",
+                                        paddingLeft: "10px",
+                                    }}>
+                                        ({this.props.groupId})
+                                    </span>
+                                </div>
+                                <div className="mx_RoomHeader_topic">
+                                    {summary.profile.short_description}
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+                    {description}
+                </div>
+            );
+        } else if (this.state.phase == "GroupView.NOT_FOUND") {
+            return (
+                <div style={{margin: "auto"}}>
+                    Group {this.props.groupId} not found
+                </div>
+            );
+        } else {
+            let extraText;
+            if (this.state.error.errcode === 'M_UNRECOGNIZED') {
+                extraText = <div>{_t('This Home server does not support groups')}</div>;
```

Comment:
I started with that and then looked at other places and concluded that Home server was slightly more prevalent but it's still a bit of a mixed bag